src/ScatterUser.test.ts

```diff
@@ -78,4 +88,95 @@ describe('ScatterUser', () => {
       }
     })
   })
+
+  describe('verifyKeyOwnership', () => {
+    it('should reject promise with an error if timeout is reached', async () => {
+      user.authenticate = jest.fn().mockImplementation((challenge, resolve) => {
+        jest.runOnlyPendingTimers();
+        setTimeout(() => {
+          resolve(challenge)
+        })
+      })
+      jest.useFakeTimers();
+
+      await expect(user.verifyKeyOwnership(challenges[0])).rejects.toThrow()
+
+      expect(user.authenticate).toHaveBeenCalledTimes(1)
+    })
+
+    it('should execute properly without an error if timeout is not reached', async () => {
+      user.authenticate = jest.fn().mockImplementation((challenge, resolve) => {
+        resolve(challenge)
+      })
+
+      const result = await user.verifyKeyOwnership(challenges[0])
+
+      expect(result).toBe(challenges[0])
+      expect(user.authenticate).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  describe('getPublicKey', () => {
+    it('should be able to get public key from challenge and signature passed in', () => {
+      const result = user.getPublicKey(challenges[0], signatureValue)
+
+      expect(result).toBe(publicKeys[0])
+    })
+
+    it('should be able to get public key from a different challenge and signature passed in', () => {
+      const result = user.getPublicKey(challenges[1], signatureValue)
```

Comment:
This reliably produces the same signatureValue?  I wouldn't think that would be the same value each time (I would expect a random factor involved).